### PR TITLE
[SPARK-20892][SparkR] Add SQL trunc function to SparkR

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?

Add SQL trunc function

## How was this patch tested?
standard test

### CODE REVIEW

**Comment by actuaryzhang** [comment]
@felixcheung @zero323

**Comment by actuaryzhang** [comment]
jenkins, retest this please

File: R/pkg/NAMESPACE [hunk 1]
```diff
@@ -357,6 +357,7 @@ exportMethods("%<=>%",
               "to_utc_timestamp",
               "translate",
               "trim",
+              "trunc",
```
Comment (felixcheung) [inline R/pkg/NAMESPACE]: it doesn't mask base::trunc?

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -4015,3 +4015,31 @@ setMethod("input_file_name", signature("missing"),
...
+#' trunc(df$c, "year")
+#' trunc(df$c, "yy")
+#' trunc(df$c, "month")
+#' trunc(df$c, "mon")
+#' }
+#' @note trunc since 2.3.0
+setMethod("trunc",
+          signature(x = "Column"),
+          function(x, format) {
+            jc <- callJStatic("org.apache.spark.sql.functions", "trunc", x@jc, format)
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: I'd `as.character(format)` here

File: R/pkg/NAMESPACE (hunk 1 above)
Comment (felixcheung) [inline R/pkg/NAMESPACE]: and looks like it has a generic already?

**Review by felixcheung — COMMENTED** [review]
minor comment, LGTM otherwise

File: R/pkg/NAMESPACE (hunk 1 above)
Comment (actuaryzhang) [inline R/pkg/NAMESPACE]: It is part of the internally S4 methods and there is already generics.
This is similar to math functions like `abs`.
https://stat.ethz.ch/R-manual/R-devel/library/base/html/groupGeneric.html

**Comment by actuaryzhang** [comment]
Added your suggested change. Thanks!

File: R/pkg/NAMESPACE (hunk 1 above)
Comment (actuaryzhang) [inline R/pkg/NAMESPACE]: And yes, it doesn't mask base. You can still do `trunc(10.5)`.

**Comment by actuaryzhang** [comment]
@felixcheung Anything else needed for this PR?

**Comment by felixcheung** [comment]
merged to master.

**Pull request closed without merging**